Adding support to avoid having the same keystore file added twice

sophoah opened the pull request:
I didn't found a very good way to obtain the ONE address without importing it first. Hence the need to create that new method DoesAddressExistTwice and remove the account from the keystore if the ONE address is found twice.

Please suggest if there is a better way to do this.
```
root@48e98e352df0:~/go/src/github.com/harmony-one/go-sdk# dist/hmy keys list
NAME                                                    ADDRESS

root@48e98e352df0:~/go/src/github.com/harmony-one/go-sdk# dist/hmy keys import-ks $(pwd)/UTC--2020-01-25T20-50-41.577244708Z--8ceda0fd494b2848fbb64b083409a7c1dcce3bfb
Imported keystore given account alias of `thunder-imported`
root@48e98e352df0:~/go/src/github.com/harmony-one/go-sdk# dist/hmy keys list
NAME                                                    ADDRESS

thunder-imported                                        one13nk6pl2ffv5y37akfvyrgzd8c8wvuwlm0g9l7l
root@48e98e352df0:~/go/src/github.com/harmony-one/go-sdk# dist/hmy keys import-ks $(pwd)/UTC--2020-01-25T20-50-41.577244708Z--8ceda0fd494b2848fbb64b083409a7c1dcce3bfb
commit: v238-878cc03-dirty, error: account one13nk6pl2ffv5y37akfvyrgzd8c8wvuwlm0g9l7l already exists
check hmy cookbook for valid examples or try adding a `--help` flag
root@48e98e352df0:~/go/src/github.com/harmony-one/go-sdk#
```

sophoah commented:
that would address https://github.com/harmony-one/harmony/issues/2163

Daniel-VDM commented:
This solution seems very heavyweight. 

Consider using our keystore's `DecryptKey` function from `github.com/harmony-one/harmony/accounts/keystore` (just add that path in the import header).

You can use it to decrypt a keystore file, get the Bech32 address (so a one1... address) and check if you can fetch anything from the keystore with that address. So the following code snippet should work:

```go
key, err := keystore.DecryptKey(keyJSON, passphrase)
if err != nil {
    return "", err
}
b32 := address.ToBech32(key.Address)
hasAddress := store.FromAddress(b32) != nil
if hasAddress {
    return "", fmt.Errorf("address %s already exists in keystore", b32)
}
```

Actually, looking at `store.FromAddress`, I see it has no documentation stating that it will return nil if nothing is found, maybe you can add that in the doc?

Also, `ks.Import` will re-decrypt the file so maybe you can just copy over the keystore file into the correct dir after you do the verification (correct passphrase & addr not in keystore).

sophoah commented:
@Daniel-VDM I fixed and tested the code successfully. 
Also added comments on a few methods defined in store/local.go
I was looking at the ks.import code and I think it is doing a little bit more than just copying the keystore file, so i didn't change the last part. Also, not sure why but the original keystore file and the one in the .hmycli after import is not exactly the same

Daniel-VDM commented on pkg/account/import.go:
```diff
@@ -88,7 +90,15 @@ func ImportKeyStore(keyPath, name, passphrase string) (string, error) {
+		return "", fmt.Errorf("address %s already exists in keystore", b32)
+	}
 	ks := store.FromAccountName(name)
 	_, err = ks.Import(keyJSON, passphrase, passphrase)
```
Import decrypts the keysJSON, which takes some time. Since we already know that the keystore file is valid at this point, we can just copy over the file to the correct directory, this will be faster.

sophoah replied:
I've tested ok the code after adding:

``` Go
uDir, _ := homedir.Dir()
path := filepath.Join(uDir, c.DefaultConfigDirName, c.DefaultConfigAccountAliasesDirName, name)	
err = os.MkdirAll(path, 0600)
if err != nil {
	return "", err
}
err = ioutil.WriteFile(filepath.Join(path, filepath.Base(keyPath)), keyJSON, 0600)
if err != nil {
	return "", err
}
```

just wondered if I should leave that in
- the import.go 
- another package like common
- a private function of account package

sophoah commented:
@Daniel-VDM pushed the changes, let me know if any other comments and feedback, thanks a lot for your guidance here
